stdlib/bool.mli

```diff
@@ -0,0 +1,65 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                         The OCaml programmers                          *)
+(*                                                                        *)
+(*   Copyright 2018 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Boolean values.
+
+    @since 4.08 *)
+
+(** {1:bools Booleans} *)
+
+type t = bool
+(** The type for boolean values. *)
+
+val not : bool -> bool
+(** [not b] is the boolean negation of [b]. *)
+
+val negate : ('a -> bool) -> ('a -> bool)
+(** [negate p v] is [not (p v)]. *)
+
+external ( && ) : bool -> bool -> bool = "%sequand"
+(** [e0 && e1] is the lazy boolean conjunction of expressions [e0] and [e1].
+    If [e0] evaluates to [false], [e1] is not evaluated. Right-associative
+    operator at precedence level 3/11. *)
+
+external ( || ) : bool -> bool -> bool = "%sequor"
+(** [e0 || e1] is the lazy boolean disjunction of expressions [e0] and [e1].
+    If [e0] evaluates to [true], [e1] is not evaluated. Right-associative
+    operator at precedence level 2/11. *)
+
+(** {1:preds Predicates and comparisons} *)
+
+val equal : bool -> bool -> bool
+(** [equal b0 b1] is [true] iff [b0] and [b1] are both either [true]
+    or [false]. *)
+
+val compare : bool -> bool -> int
+(** [compare b0 b1] is a total order on boolean values. [false] is smaller
+    than [true]. *)
+
+(** {1:convert Converting} *)
+
+val to_int : bool -> int
```

Comment:
How natural/common are to_int/to_float?  They seem to be a bit ad hoc.  Do you have the intuition that those definitions are commonly used?

Reply:
I certainly did an inlined form of `to_int` in the context of C bindings. `to_float` maybe not I can remove it if it's desired.

Reply:
In the context of C bindings, wouldn't `of_int` (mapping zero to false and any other integer to true) be equally useful?

Reply:
> In the context of C bindings, wouldn't `of_int` (mapping zero to false and any other integer to true) be equally useful?

I'd say: it depends this is why I didn't add it.